modules/age.nix

```
@@ -103,9 +116,30 @@ in
       message = "age.sshKeyPaths must be set.";
     }];
 
+    # Create a new directory full of secrets for symlinking (this helps
+    # ensure removed secrets are actually removed, or at least become
+    # invalid symlinks).
+    system.activationScripts.agenixMountSecrets = ''
+      _count="$(basename "$(readlink /run/secrets)" || echo 0)"
+      (( ++_count ))
+      echo "[agenix] symlinking new secrets generation $_count to /run/secrets..."
+      mkdir -p "${cfg.secretsMountPoint}"
+      chmod 0750 "${cfg.secretsMountPoint}"
+      mount | grep "${cfg.secretsMountPoint} type ramfs" -q || mount -t ramfs none "${cfg.secretsMountPoint}" -o nodev,nosuid,mode=0750
+      mkdir -p "${cfg.secretsMountPoint}/$_count"
+      chmod 0750 "${cfg.secretsMountPoint}/$_count"
+      chown :keys "${cfg.secretsMountPoint}" "${cfg.secretsMountPoint}/$_count"
+      ln -sfn "${cfg.secretsMountPoint}/$_count" /run/secrets
+      echo "[agenix] removing old secrets generation $((( _count - 1 )))..."
```

Changing this to hopefullly read `removing old secrets (generation X)`, which IMO improves clarity. Haven't tested this though.

Also, you don't need the additional brackets for `$`: `$(( _count - 1 ))` instead of `$((( _count - 1 )))`.

```suggestion
      echo "[agenix] removing old secrets (generation $(( _count - 1 )))..."
```